### PR TITLE
Fixes for Hausdorff distance and some ambiguities

### PR DESCRIPTION
**Pull request by mateuszbaran**
Most of the changes here are due to tests failing on master so I tried to make some minimal fixes that make all tests pass. I don't quite like adding new methods to `result_type` here and that was originally outside of the scope of this PR but that seems better than leaving the tests failing.

### CODE REVIEW

File: src/hausdorff.jl [hunk 1]
```diff
@@ -9,7 +9,13 @@ struct MaxReduction  <: ReductionOperation end
...
 
 _reduce(op::MaxReduction, x)  = maximum(x)
-_reduce(op::MeanReduction, x) = sum(x) / length(x)
+function _reduce(op::MeanReduction, x)
+    if length(x) == 0
+        return zero(eltype(x))
+    else
+        return sum(x) / length(x)
+    end
+end
```
Comment (johnnychen94) [inline src/hausdorff.jl:18]: curious to ask, is this equivalent to `mean(x)`? I'm not very sure if it should be `0` when `length(x) == 0` here.

**Review by johnnychen94 — APPROVED** [review]

File: src/hausdorff.jl (hunk 1 above)
Comment (mateuszbaran) [reply src/hausdorff.jl:18]: Thanks for noticing and sorry, I think I checked that against `Statistics.mean` but now I checked again and it actually returns `NaN` as a mean of an empty collection. I'll update it.

**Comment by johnnychen94** [comment]
@mateuszbaran Thank you for the fixes; this is of a great help especially that we don't have many contributors!

**Comment by mateuszbaran** [comment]
No problem! It's also nice to quickly have feedback on my patches :slightly_smiling_face: .

**Pull request merged**